### DISCUSSION TITLE
Sync Azure SSO Groups to Netbox

### DISCUSSION THREAD

**Discussion by salman-uk** [netbox-community/netbox#18205] [Ideas]
Hi,

Is is possible to sync Azure AD SSO groups to Netbox on the local Netbox instance in a similar way as described for Netbox Cloud version here:

https://www.youtube.com/watch?v=Vg0xpWJiKAs

This will be for assigning permissions by groups as shown in the video.

Kind regards,

Salman

**Comment by salman-uk** [comment]
Any thoughts?

**Comment by ALL-SPACE-Anas** [comment]
@salman-uk I'm looking for the same, please let me know if you found something.

**Comment by salman-uk** [comment]
Not yet, if one of the Netbox team can look at this and reply that would be great.

**Comment by CTV-2023** [comment]
With saml and a custom pipeline yes you can. A little search here or on reddit will help you ;)

**Comment by divad1196** [comment]
I rejoin @CTV-2023, we also did it with a custom AUTH pipeline.
Since your users will most probably be authenticated from the AD as well, it covers your needs.

You can look at the issue #9216 
NOTE: The `SOCIAL_AUTH_PIPELINE` was changed in the meantime for security reason, so use the new one. but the `set_role` is mostly fine.

**Comment by ALL-SPACE-Anas** [comment]
@CTV-2023 @divad1196 thanks folks, I got it to "work" following https://github.com/netbox-community/netbox/discussions/13129 and https://github.com/netbox-community/netbox/discussions/9216. I still have issue where the response JSON from azure doesn't include the "roles" key for some reason but looks like this is an azure issue rather a netbox one.
